### PR TITLE
Ports /tg/ emote datums

### PR DESCRIPTION
**Pull request by DominikPanic**
:cl: LaKiller8
code: /tg/ emote datums have been ported.
/:cl:

### CODE REVIEW

File: code/datums/emotes.dm [hunk 1]
```diff
@@ -0,0 +1,169 @@
+#define EMOTE_VISIBLE 1
+#define EMOTE_AUDIBLE 2
+
+/datum/emote
+	var/key = "" //What calls the emote
+	var/key_third_person = "" //This will also call the emote
+	var/message = "" //Message displayed when emote is used
+	var/message_alien = "" //Message displayed if the user is a grown alien
```
Comment (psykzz) [inline code/datums/emotes.dm]: can this be `message_overrides = list()` instead and pass an associate list of string type to message?

Comment (DominikPanic) [reply code/datums/emotes.dm]: I am just porting /tg/ code at the moment, feel free to improve it later but this is out of scope

File: code/datums/emotes.dm [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	var/message_monkey = "" //Message displayed if the user is a monkey
+	var/message_simple = "" //Message to display if the user is a simple_animal
+	var/message_param = "" //Message to display if a param was given
+	var/emote_type = EMOTE_VISIBLE //Whether the emote is visible or audible
+	var/restraint_check = FALSE //Checks if the mob is restrained before performing the emote
+	var/muzzle_ignore = FALSE //Will only work if the emote is EMOTE_AUDIBLE
+	var/list/mob_type_allowed_typecache = /mob //Types that are allowed to use that emote
+	var/list/mob_type_blacklist_typecache //Types that are NOT allowed to use that emote
+	var/list/mob_type_ignore_stat_typecache
+	var/stat_allowed = CONSCIOUS
```
Comment (psykzz) [inline code/datums/emotes.dm]: Can this and restraint check be refactored into bitflags?

Comment (DominikPanic) [reply code/datums/emotes.dm]: Again, out of scope.

Comment (spookydonut) [reply code/datums/emotes.dm]: in-scope imo

Comment (Rohesie) [reply code/datums/emotes.dm]: `stat_allowed` conversion to bitflags wouldn't be that clean, it's not a boolean value.
`restraint_check`, `muzzle_ignore`, `vary` and `only_forced_audio` would make more sense.

Comment (DominikPanic) [reply code/datums/emotes.dm]: 1. Losing compatibility with /tg/
2. Look at the previous emote code. This is 10000x better.

Comment (DominikPanic) [reply code/datums/emotes.dm]: Though I'll rather pay this ransom than let this PR sit for a week since it's actively hurting me. You win again.

Comment (DominikPanic) [reply code/datums/emotes.dm]: > 
> 
> Can this and restraint check be refactored into bitflags?

stat is not a bitflag

**Pull request merged**